pynest/nest/lib/hl_api_simulation.py

```diff
@@ -382,11 +382,31 @@ def SetKernelStatus(params):
     GetKernelStatus
 
     """
-
+    for key in params.keys():
+        readonly = _sks_params.get(key)
+        if readonly is None:
+            # If the parameter is not in the docstring
+            raise KeyError(f'`{key}` is not a valid kernel parameter, '
+                           'valid parameters are: '
+                           ', '.join(f"'{p}'" for p in _sks_params.keys()))
+        elif readonly:
+            # If the parameter is tagged as read only
+            raise KeyError(f'`{key}` is a read only parameter and cannot '
+                           'be defined using SetKernelStatus')
     sps(params)
     sr('SetKernelStatus')
 
 
+# Parse the `SetKernelStatus` docstring to obtain all valid and readonly params
+lines = SetKernelStatus.__doc__.split('\n')
+# Get the lines describing parameters
+param_lines = (line for line in lines if ' : ' in line)
```

Comment:
To prevent parsing the `Returns` as parameters we should stop parsing at `**Output**`:

```python
# Get the lines describing parameters, stop parsing lines at `Output`
param_lines = iter((line.strip() for line in lines if ' : ' in line).__next__, "**Output**")
```

Reply:
Done